app/src/apps/commander/Play.tsx

```diff
@@ -0,0 +1,241 @@
+                                : setUnit({ unit, x, y })
+                        }
+                        onPoint={(unit, x, y) => {
+                            performSelectedAction(unit, x, y).then(() =>
+                                setWait(false),
+                            );
+                        }}
+                    />
+                </div>
+            </div>
+            <div className="flex flex-col justify-center">
+                <h2 className="my-2 text-xl">Game</h2>
+                <p className="">
+                    <button
+                        onClick={() => destroy().then(() => setWait(false))}
+                    >
+                        Destroy Game
+                    </button>
+                </p>
+                <p className="mb-4">
+                    <button
+                        onClick={() => nextTurn().then(() => setWait(false))}
+                    >
+                        Next Turn ({game.turn + 1})
+                    </button>
+                </p>
+                <UnitStats
+                    game={game}
+                    unit={unit?.unit || null}
+                    onSelect={(idx, action) => selectAction({ idx, action })}
+                />
+            </div>
+        </div>
+    );
+
+    async function performSelectedAction(
+        _unit: typeof Unit.$inferType | null,
+        x: number,
+        y: number,
+    ) {
+        if (!zkLogin.address) return;
+        if (!action) return;
+        if (!unit) return;
+        if (!game) return;
+        if (wait) return;
+
+        let timer: any;
+
+        setWait(true);
+
+        if (game.turn > unit.unit.turn) {
+            unit.unit.turn = game.turn;
+            unit.unit.ap.value = unit.unit.ap.maxValue;
+            setUnit({ ...unit });
+        }
+
+        // very silly check for now
+        if (action.action.inner.$kind === "Move") {
+            const maxRange = unit.unit.ap.value / action.action.cost;
+            const range = Math.abs(unit.x - x) + Math.abs(unit.y - y);
+
+            if (game.map.grid[x][y].$kind !== "Empty")
+                return console.log("not empty");
+
+            console.log(range, maxRange);
+
+            if (range > maxRange) return console.log("out of range");
+
+            const inspect = new Transaction();
+            inspect.moveCall({
+                target: `${packageId}::commander::trace_path`,
+                arguments: [
+                    inspect.object(normalizeSuiObjectId(game.id)),
+                    inspect.pure.u16(maxRange),
+                    inspect.pure.u16(unit.x),
+                    inspect.pure.u16(unit.y),
+                    inspect.pure.u16(x),
+                    inspect.pure.u16(y),
+                ],
+            });
+
+            const result = await client.devInspectTransactionBlock({
+                sender: zkLogin.address!,
+                transactionBlock: inspect as any,
```

Comment:
why does this need casting to `any`?

Reply:
React version of `Transaction` type is of different version of `@mysten/sui` one. Results in type error. Same happens with Enoki types btw.